src/incflo_advance.cpp

```diff
@@ -147,110 +140,101 @@ void incflo::ApplyPredictor (bool incremental_projection)
+    auto tra_forces = m_repo.create_scratch_field(m_ntrac, nghost_force());
+    auto tra_eta = m_repo.create_scratch_field(m_ntrac, 1);
 
     // *************************************************************************************
     // Define the forcing terms to use in the Godunov prediction
     // *************************************************************************************
     if (m_use_godunov)
     {
-        compute_vel_forces(GetVecOfPtrs(vel_forces), get_velocity_old_const(),
-                           get_density_old_const(),
-                           get_tracer_old_const());
+        compute_vel_forces((*vel_forces).vec_ptrs(),
+                           velocity_old.vec_const_ptrs(),
+                           density_old.vec_const_ptrs(),
+                           tracer_old.vec_const_ptrs());
 
         // Note this is forcing for (rho s), not for s
         if (m_advect_tracer)
-           compute_tra_forces(GetVecOfPtrs(tra_forces), get_density_old_const());
+           compute_tra_forces((*tra_forces).vec_ptrs(), density_old.vec_const_ptrs());
     }
 
     // *************************************************************************************
     // Compute viscosity / diffusive coefficients
     // *************************************************************************************
-    compute_viscosity(GetVecOfPtrs(vel_eta), GetVecOfPtrs(tra_eta),
-                      get_density_old_const(), get_velocity_old_const(), get_tracer_old_const(),
+    compute_viscosity((*vel_eta).vec_ptrs(), (*tra_eta).vec_ptrs(),
+                      density_old.vec_const_ptrs(), velocity_old.vec_const_ptrs(), tracer_old.vec_const_ptrs(),
                       m_time.current_time(), 1);
 
     // *************************************************************************************
     // Compute explicit viscous term
     // *************************************************************************************
     if (need_divtau()) {
-        get_diffusion_tensor_op()->compute_divtau(get_divtau_old(),
-                                                  get_velocity_old_const(),
-                                                  get_density_old_const(),
-                                                  GetVecOfConstPtrs(vel_eta));
-        if (m_godunov_include_diff_in_forcing)
-            for (int lev = 0; lev <= finest_level; ++lev)
-                MultiFab::Add(vel_forces[lev], m_leveldata[lev]->divtau_o, 0, 0, AMREX_SPACEDIM, 0);
+        get_diffusion_tensor_op()->compute_divtau(divtau.vec_ptrs(),
+                                                  velocity_old.vec_const_ptrs(),
+                                                  density_old.vec_const_ptrs(),
+                                                  (*vel_eta).vec_const_ptrs());
+        if (m_use_godunov)
+            amr_wind::field_ops::add(*vel_forces, divtau, 0, 0, AMREX_SPACEDIM, 0);
```

Comment:
So I see that you're using `divtau` instead of `divtau_old` here, do you know what was the reason to have old/new states for `divtau` and `laplacian scalar`?

Reply:
I used the old state above but didn't label it with _old, the same goes for some other variables. I could add if that is more clear. Godunov will only need one state, MOL needs both states of divtau (and laps)

Reply:
No, I saw your comment about `_old` for velocity, etc. but somehow didn't catch that for `divtau`. What you have is good.